src/misc.c

```diff
@@ -162,15 +162,13 @@ char *fgetll(char *s, size_t size, FILE *stream)
 			ungetc(c, stream);
 		return s;
 	}
-	cp = s;
+	cp = strdup(s);
+
 	while (1) {
-		cp2 = cp;
-		cp = mem_alloc(len+60000);
-		strcpy(cp, cp2);
-		if (cp2 != s)
-			MEM_FREE(cp2);
+		cp = realloc(cp, 2 * len);
+
```

Comment:
cp = realloc(cp,...)  is BAD practice.  Read the info about realloc. It is REALLY a hairy function.  You should never realloc a pointer using the same pointer.  Always use a different pointer (or save the original one off).